agent/structs/config_entry_routes.go

```diff
@@ -135,3 +163,13 @@ func (e *TCPRouteConfigEntry) GetEnterpriseMeta() *acl.EnterpriseMeta {
 	}
 	return &e.EnterpriseMeta
 }
+
+// TCPService is a service reference for a TCPRoute
+type TCPService struct {
+	Name string
+	// Weight is an arbitrary integer used in calculating how much
+	// traffic should be sent to the given service.
```

Comment:
I think it'd be helpful to explain how `Weight` works here. The Gateway API [docs](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.BackendRef) say
> Weight specifies the proportion of requests forwarded to the referenced backend. This is computed as weight/(sum of all weights in this BackendRefs list)

Since `tcp-route` can only route to a single service right now, this is just a placeholder for future functionality, right?